src/components/Fieldset/DynamicFieldset/DynamicFieldset.types.ts

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+
+type BaseProps = {
```

Comment:
To avoid repeating all props here, you can import and inherit from `FieldsetProps` as both shared the same based props